com.sap.cloud.lm.sl.cf.process/src/test/java/com/sap/cloud/lm/sl/cf/process/steps/UpdateServiceKeysStepTest.java

```diff
@@ -1,140 +1,150 @@
-        MethodExecution<String> methodExec;
-        switch (stepPhase) {
-            case POLLING:
-                methodExec = new MethodExecution<String>(null, ExecutionState.FINISHED);
-                break;
-            case STEP_EXECUTION:
-                methodExec = new MethodExecution<String>(null, ExecutionState.EXECUTING);
-                break;
-            default:
-                throw new UnsupportedOperationException("Unsupported test phase");
-        }
-        Mockito.when(serviceUpdater.updateServiceTags(Matchers.any(), Matchers.any(), Matchers.any()))
-               .thenReturn(methodExec);
+    private List<CloudServiceKey> buildServiceKeys(List<String> serviceKeysNames, List<String> updatedServiceKeys,
+                                                   CloudServiceExtended service) {
+        return serviceKeysNames.stream()
+                               .map(serviceKeyName -> buildCloudServiceKey(service, updatedServiceKeys, serviceKeyName))
+                               .collect(Collectors.toList());
     }
 
-    @Test
-    public void testExecute() throws Exception {
-        prepareResponses(STEP_EXECUTION);
-        step.execute(context);
-        assertStepPhase(STEP_EXECUTION);
+    private void prepareContext(List<CloudServiceKey> serviceKeys, boolean canDeleteServiceKeys, CloudServiceExtended service) {
+        StepsUtil.setServiceKeysToCreate(context, MapUtil.asMap(SERVICE_NAME, serviceKeys));
+        context.setVariable(Constants.PARAM_DELETE_SERVICE_KEYS, canDeleteServiceKeys);
+        StepsUtil.setServiceToProcess(service, context);
+
+    }
 
-        if (getExecutionStatus().equals("DONE")) {
-            return;
+    private ImmutableCloudServiceKey buildCloudServiceKey(CloudServiceExtended service, List<String> updatedServiceKeys,
+                                                          String serviceKeyName) {
+        if (updatedServiceKeys.contains(serviceKeyName)) {
+            return ImmutableCloudServiceKey.builder()
+                                           .name(serviceKeyName)
+                                           .service(service)
+                                           .parameters(MapUtil.asMap("name", "new-value"))
+                                           .build();
         }
-        assertMethodCalls();
+        return ImmutableCloudServiceKey.builder()
+                                       .name(serviceKeyName)
+                                       .service(service)
+                                       .build();
     }
 
-    private void assertMethodCalls() {
-        Mockito.verify(serviceUpdater, Mockito.times(1))
-               .updateServiceTags(Matchers.any(), Matchers.any(), Matchers.any());
+    private void prepareServiceOperationExecutor(List<CloudServiceKey> existingServiceKeys) {
+        when(serviceOperationExecutor.executeServiceOperation(any(), Matchers.<Supplier<List<CloudServiceKey>>> any(),

```

Comment:
Is there another executeServiceOperation method with the same amount of arguments for this casting to be needed?
If not, this is excessive and hinders reading

Reply:
Yes, ServiceOperationExecutor has two methods with the same amount of arguments.